lib/src/resources/movie_api_provider.dart

```diff
@@ -18,4 +18,21 @@ class MovieApiProvider {
       throw Exception('Failed to load post');
     }
   }
+
+  Future<Movie> fetchMoviesBySearch(String query) async {
+    final response = await client.get(
+      Uri.parse(
+        StringConstants.uriPathSearch +
+            query +
+            StringConstants.uriPathSearchQuery,
+      ),
+    );
+    if (response.statusCode == 200) {
+      // If the call to the server was successful, parse the JSON
+      return Movie.fromJson(json.decode(response.body));
+    } else {
+      // If that call was not successful, throw an error.
```

Comment:
it's good to have comments explaining difficult implementations that you can elaborate on the work done, but in this case where you check if the response is successful or not, it's self explanatory. You can remove the comments

Reply:
Fixed